components/eamxx/src/physics/nudging/eamxx_nudging_process_interface.cpp

```diff
@@ -27,329 +41,271 @@ void Nudging::set_grids(const std::shared_ptr<const GridsManager> grids_manager)
-                                           T_mid,
-                                           m_num_src_levs,
-                                           m_num_levs);
 
-  perform_vertical_interpolation<Real,1,2>(p_mid_ext_p,
-                                           p_mid,
-                                           qv_ext_p,
-                                           qv,
-                                           m_num_src_levs,
-                                           m_num_levs);
-
-  perform_vertical_interpolation<Real,1,2>(p_mid_ext_p,
-                                           p_mid,
-                                           u_ext_p,
-                                           u,
-                                           m_num_src_levs,
-                                           m_num_levs);
-
-  perform_vertical_interpolation<Real,1,2>(p_mid_ext_p,
-                                           p_mid,
-                                           v_ext_p,
-                                           v,
-                                           m_num_src_levs,
-                                           m_num_levs);
-
-  //This code removes any masked values and sets them to the corresponding
-  //values at the lowest/highest pressure levels from the external file
-  auto T_mid_ext = m_fields_ext["T_mid"];
-  auto p_mid_ext = m_fields_ext["p_mid"];
-  auto qv_ext = m_fields_ext["qv"];
-  auto u_ext = m_fields_ext["u"];
-  auto v_ext = m_fields_ext["v"];
-
-  const int num_cols = T_mid.extent(0);
-  const int num_vert_packs = T_mid.extent(1);
-  const int num_vert_packs_ext = T_mid_ext.extent(1);
-  const auto policy = ESU::get_default_team_policy(num_cols, num_vert_packs);
-  Kokkos::parallel_for("correct_for_masked_values", policy,
-     	       KOKKOS_LAMBDA(MemberType const& team) {
+  // Perform time interpolation
+  m_time_interp.perform_time_interpolation(ts);
+
+  // Process data and nudge the atmosphere state
+  const auto& p_mid_v     = get_field_in("p_mid").get_view<const mPack**>();
+  view_Nd<mPack,2> p_mid_ext_p;
+  view_Nd<mPack,1> p_mid_ext_1d;
+  if (m_src_pres_type == DYNAMIC) {
+    const auto& p_mid_ext   = get_helper_field("p_mid_ext").get_view<mPack**>();
+    p_mid_ext_p = view_Nd<mPack,2>(reinterpret_cast<mPack*>(p_mid_ext.data()),
+  				     m_num_cols,m_num_src_levs);
+  } else if (m_src_pres_type == STATIC) {
+    p_mid_ext_1d   = get_helper_field("p_mid_ext").get_view<mPack*>();
+  }
+  for (auto name : m_fields_nudge) {
+    auto atm_state_field = get_field_out(name);
+    auto int_state_field = get_helper_field(name);
+    auto ext_state_field = get_helper_field(name+"_ext").get_view<mPack**>();
+    auto atm_state_view  = atm_state_field.get_view<mPack**>();  // TODO: Right now assume whatever field is defined on COLxLEV
+    auto int_state_view  = int_state_field.get_view<mPack**>();
+    view_Nd<mMask,2> int_mask_view("",m_num_cols,m_num_levs); // Track mask of interpolated values
```

Comment:
We should avoid allocating views inside the run method. It has a non trivial cost.